setup/plugins/installFiles/l10nDownload.civi-setup.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * @file
+ *
+ * Generate the civicrm.settings.php file.
+ */
+
+if (!defined('CIVI_SETUP')) {
+  exit("Installation plugins must only be loaded by the installer.\n");
+}
+
+/**
+ * Validate the $model.
+ */
+\Civi\Setup::dispatcher()
+  ->addListener('civi.setup.checkRequirements', function(\Civi\Setup\Event\CheckRequirementsEvent $e) {
+    $lang = \Civi\Setup::instance()->getModel()->lang;
+    if ($lang && $lang != 'en_US') {
+      // Use CiviCRM Core directory as a fall back.
+      $baseDir = $e->getModel()->srcPath;
+      if (isset($e->getModel()->paths['civicrm.private']['path'])) {
+        // If the civicrm files directory is set use this as the base path.
+        $baseDir = $e->getModel()->paths['civicrm.private']['path'];
+      }
+      // Set l10n basedir as a define. The GenCode.php tries to locate the l10n files
+      // from this location if other than l10n in the civicrm core directory.
+      if (!isset($e->getModel()->paths['civicrm.l10n']['path'])) {
+        if (defined('CIVICRM_L10N_BASEDIR')) {
```

Comment:
How would I `define` this? In drupal settings.php? Maybe use `CRM_Utils_Constant::value()` and then I can set it with an environment variable.

Reply:
@demeritcowboy I had another look and changed it to use the `CRM_Utils_Constant`. I also changed setting the define to setting the environment variable.